### PR TITLE
#908: include correlationID into error response

### PR DESCRIPTION
**Pull request by kaisong1990**
## Proposed changes

Describe what this PR is trying to do.

## Type of change

- [ ] Feature work
- [x] Bug fix
- [ ] Documentation
- [ ] Engineering change
- [ ] Test
- [ ] Logging/Telemetry

## Risk

- [ ] High – Errors could cause MAJOR regression of many scenarios. (Example: new large features or high level infrastructure changes)
- [ ] Medium – Errors could cause regression of 1 or more scenarios. (Example: somewhat complex bug fixes, small new features)
- [x] Small – No issues are expected. (Example: Very small bug fixes, string changes, or configuration settings changes)

### CODE REVIEW

**Comment by kaisong1990** [comment]
> Please update MSALErrorConverterTests

UTests have been added

File: MSAL/src/MSALErrorConverter.m [hunk 1]
```diff
@@ -201,6 +212,7 @@ + (NSError *)errorWithDomain:(NSString *)domain
         msalUserInfo[mappedKey] = userInfo[key];
     }
 
+    if (!msalUserInfo[MSIDCorrelationIdKey] && correlationId) msalUserInfo[MSIDCorrelationIdKey] = correlationId;
```
Comment (oldalton) [inline MSAL/src/MSALErrorConverter.m:215]: should this be MSALCorrelationIDKey and not MSIDCorrelationIdKey?

File: MSAL/test/unit/MSALErrorConverterTests.m [hunk 2]
```diff
@@ -404,5 +404,51 @@ - (void)testErrorConversion_whenErrorMappedToInternalError_shouldSetInternalErro
...
+                                                     code:MSIDErrorInternal
+                                                userInfo:@{MSIDCorrelationIdKey:uuid}];
+    
+    NSError *msalError = [MSALErrorConverter msalErrorFromMsidError:msidError
+                                                     classifyErrors:YES
+                                                 msalOauth2Provider:nil
+                                                      correlationId: nil];
+    
+    XCTAssertNotNil(msalError.userInfo[MSIDCorrelationIdKey]);
+    XCTAssertEqual(msalError.userInfo[MSIDCorrelationIdKey], uuid);
```
Comment (oldalton) [inline MSAL/test/unit/MSALErrorConverterTests.m:420]: this should be MSALCorrelationIDKey

File: MSAL/src/MSALErrorConverter.m (hunk 1 above)
Comment (oldalton) [inline MSAL/src/MSALErrorConverter.m:215]: do we want to save it as a string and not UUID? if anyone tries to serialize error, UUID would fail I believe

Comment (kaisong1990) [reply MSAL/src/MSALErrorConverter.m:215]: Yes, I messed with those two keys

File: MSAL/test/unit/MSALErrorConverterTests.m (hunk 2 above)
Comment (kaisong1990) [reply MSAL/test/unit/MSALErrorConverterTests.m:420]: Will update

File: MSAL/src/MSALErrorConverter.m (hunk 1 above)
Comment (kaisong1990) [reply MSAL/src/MSALErrorConverter.m:215]: Just read the code a little bit and notice we are using string format already in msid. Will make the changes

File: MSAL/src/MSALErrorConverter.m [hunk 3]
```diff
@@ -135,14 +135,25 @@ + (NSError *)msalErrorFromMsidError:(NSError *)msidError
...
+                            correlationId:(NSUUID *)correlationId
 {
     return [self errorWithDomain:msidError.domain
                             code:msidError.code
                 errorDescription:msidError.userInfo[MSIDErrorDescriptionKey]
                       oauthError:msidError.userInfo[MSIDOAuthErrorKey]
                         subError:msidError.userInfo[MSIDOAuthSubErrorKey]
                  underlyingError:msidError.userInfo[NSUnderlyingErrorKey]
-                   correlationId:msidError.userInfo[MSIDCorrelationIdKey]
+                   correlationId:msidError.userInfo[MSIDCorrelationIdKey] ? : correlationId
```
Comment (oldalton) [inline MSAL/src/MSALErrorConverter.m:156]: is msidError.userInfo[MSIDCorrelationIdKey] also NSUUID type? Make sure both options are of the same type.

File: MSAL/src/MSALErrorConverter.m [hunk 4]
```diff
@@ -201,6 +212,7 @@ + (NSError *)errorWithDomain:(NSString *)domain
         msalUserInfo[mappedKey] = userInfo[key];
     }
 
+    if (!msalUserInfo[MSALCorrelationIDKey] && correlationId) msalUserInfo[MSALCorrelationIDKey] = correlationId.UUIDString;
```
Comment (oldalton) [inline MSAL/src/MSALErrorConverter.m:215]: I believe we can now remove __unused tag above

Comment (kaisong1990) [reply MSAL/src/MSALErrorConverter.m:215]: Do you mean the method signature? If yes, I will remove it and make sure the type is correct

**Pull request merged**